HealthcareApp/src/Healthcare.Infrastructure/Repository/TokenGeneratorService.cs

```diff
@@ -0,0 +1,54 @@
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using Application.Abstractions;
+using Application.DTOs.Login;
+using Healthcare.Infrastructure.Persistance;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Infrastructure.Repository;
+
+public class TokenGeneratorService : ITokenGeneratorService
+{
+    private readonly UserManager<ApplicationUser> _userManager;
+    private readonly IConfiguration _configuration;
+
+    public TokenGeneratorService(UserManager<ApplicationUser> userManager, IConfiguration configuration)
```

Comment:
Use Options pattern instead of injecting `IConfiguration`. If you inject IConfiguration, then your class can access all data from the configuration, but your TokenService should not know about all the data from configuration, it's not his responsibility. Options pattern in asp net will resolve your problem, please read about it